src/backends/torch/torchsolver.cc

```diff
@@ -201,29 +201,41 @@ namespace dd
 
   int TorchSolver::resume(const APIData &ad_mllib, const TorchModel &mlmodel,
                           const torch::Device &main_device,
-                          double &best_metric_value,
-                          int64_t &best_iteration_number)
+                          std::vector<double> &best_metric_values,
+                          std::vector<int64_t> &best_iteration_numbers,
+                          const std::vector<std::string> &set_names)
   {
     // reload solver if asked for and set it value accordingly
     if (ad_mllib.has("resume") && ad_mllib.get("resume").get<bool>())
       {
-        std::string bestfilename
-            = mlmodel._repo + mlmodel._best_model_filename;
-        std::ifstream bestfile;
-        try
-          {
-            std::string tmp;
-            std::string bin;
-            bestfile.open(bestfilename, std::ios::in);
-            // first three fields are thrown away
-            bestfile >> tmp >> bin >> tmp >> tmp;
-            bestfile.close();
-            best_iteration_number = std::atof(bin.c_str());
-            best_metric_value = std::atof(tmp.c_str());
-          }
-        catch (std::exception &e)
+        for (size_t test_id = 0; test_id < best_iteration_numbers.size();
+             ++test_id)
           {
-            _logger->info("no previous best model file");
+            std::string bestfilename
+                = mlmodel._repo
+                  + fileops::insert_suffix("_test_" + std::to_string(test_id),
+                                           mlmodel._best_model_filename);
+            std::ifstream bestfile;
+            try
+              {
+                std::string tmp;
+                std::string bin;
+                std::string test_name;
+                bestfile.open(bestfilename, std::ios::in);
+                // first three fields are thrown away
+                bestfile >> tmp >> bin >> tmp >> tmp >> tmp >> test_name;
+                if (test_name != set_names[test_id])
+                  _logger->warn(
+                      "test names not matching: {} (API) vs {} (file)",
+                      set_names[test_id], test_name);
+                bestfile.close();
+                best_iteration_numbers[test_id] = std::atof(bin.c_str());
```

Comment:
How do we choose checkpoint to load from eventually when we have multiple metrics ?

Reply:
API parameter? It could be interesting to keep all checkpoint and compare them with eachother

Reply:
Currently I think the most recent checkpoint is used

Reply:
indeed it is the most recent file that is used, this is the behavior defined here https://github.com/jolibrain/deepdetect/blob/dbafe3a315a3f8f5920e9d7b8f77da76e6014251/src/backends/torch/torchmodel.cc#L54

ATM, several weights files remains in the repo: the best for every metric, and the regular snapshots. 

for now, user has to copy the proper file into  new repo (or equivalently remove unwanted weights). 
upgrading this behavior (for instance changing resume from true to best for metric n) may be out of the scope of this PR